packages/devtools_app/lib/src/ui/flutter/flutter_icon_renderer.dart

```diff
@@ -0,0 +1,140 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// Library to render [DevToolsIcon]s as [Widget]s.
+library icon_renderer;
+
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+import '../icons.dart';
+import '../material_icons.dart';
+import '../theme.dart';
+
+final Expando<Widget> widgetExpando = Expando('IconRenderer');
+
+typedef DrawIconImageCallback = void Function(Canvas canvas);
+
+String _rewriteIconAssetPath(String path) {
+  assert(path.startsWith('/'));
+  // Paths under flutter will include the web directory while paths for the
+  // legacy dart:html app do not.
+  path = 'web$path';
+  if (path.endsWith('.png') && !path.endsWith('@2x.png')) {
+    // By convention icons all have high DPI verisons with @2x added to the
+    // file name.
+    // We could opt out of the @2x versions for low DPI devices but there isn't
+    // really much harm in using the higher resolution icons on lower DPI
+    // displays.
+    return '${path.substring(0, path.length - 4)}@2x.png';
+  }
+  return path;
+}
+
+class _ColorIconPainter extends CustomPainter {
+  const _ColorIconPainter(this.icon);
+
+  final ColorIcon icon;
+
+  static const double iconMargin = 1;
+
+  Color get color => icon.color;
+
+  @override
+  void paint(Canvas canvas, Size size) {
+    // draw a black and gray grid to use as the background to disambiguate
+    // opaque colors from translucent colors.
+    final greyPaint = Paint()..color = grey;
+    final iconRect = Rect.fromLTRB(
+      iconMargin,
+      iconMargin,
+      size.width - iconMargin,
+      size.height - iconMargin,
+    );
+    canvas
+      ..drawRect(
+        Rect.fromLTRB(iconMargin, iconMargin, size.width - iconMargin,
+            size.height - iconMargin),
+        Paint()..color = defaultBackground,
+      )
+      ..drawRect(
+          Rect.fromLTRB(
+              iconMargin, iconMargin, size.width * 0.5, size.height * 0.5),
```

Comment:
if we keep `size.width - iconMargin` above, shouldn't this be `(size.width - iconMargin) * 0.5`? Otherwise the resulting rect will not be exactly half the size of the earlier drawn one. Same for height and same for the next rect we draw.

Reply:
this math also works because it is LTRB instead of using width and height. The center of the icon is still at exactly size.width * 0.5 and size.height * 0.5 so this works nicely.